res/controllers/Traktor-Kontrol-S3-hid-scripts.js

```diff
@@ -60,2149 +96,2160 @@ TraktorS3.ChannelColors = {
+
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter3", 0x1F, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter2", 0x21, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel3]_Effect1]", "parameter1", 0x23, 0xFFFF, this.parameterHandler);
+
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter3", 0x31, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter2", 0x33, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[EqualizerRack1_[Channel4]_Effect1]", "parameter1", 0x35, 0xFFFF, this.parameterHandler);
+
+        this.registerInputScaler(messageLong, "[Master]", "crossfader", 0x0B, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[Master]", "gain", 0x17, 0xFFFF, TraktorS3.Controller.prototype.masterGainHandler.bind(this));
+        this.registerInputScaler(messageLong, "[Master]", "headMix", 0x1D, 0xFFFF, this.parameterHandler);
+        this.registerInputScaler(messageLong, "[Master]", "headGain", 0x1B, 0xFFFF, this.parameterHandler);
+
+        this.hid.registerInputPacket(messageLong);
+
+        for (const ch in this.Channels) {
+            const chanob = this.Channels[ch];
+            engine.makeConnection(ch, "playposition",
+                TraktorS3.Channel.prototype.playpositionChanged.bind(chanob));
+            engine.connectControl(ch, "track_loaded",
+                TraktorS3.Channel.prototype.trackLoadedHandler.bind(chanob));
+            engine.connectControl(ch, "end_of_track",
+                TraktorS3.Channel.prototype.endOfTrackHandler.bind(chanob));
+        }
 
-    // Represents the first-pressed deck switch button, used for tracking deck clones.
-    this.deckSwitchPressed = "";
+        // Query the current values from the controller and set them. The packet
+        // parser ignores the first time a value is set, so we'll need to set it
+        // with different values once. Report 2 contains the state of the mixer
+        // controls.
+        const report2Values = new Uint8Array(controller.getInputReport(2));
+        TraktorS3.incomingData(new Uint8Array([2, ...Uint8Array.from(report2Values.map(x => ~x))]));
+        TraktorS3.incomingData(new Uint8Array([2, ...Uint8Array.from(report2Values)]));
+
+        // Report 1 is the state of the deck controls. These shouldn't have any
+        // initial effect, and most of these values will be 0 anyways. We'll
+        // just tell the packet parser the current values so it won't ignore the
+        // next input.
+        const report1Values = new Uint8Array(controller.getInputReport(1));
+        TraktorS3.incomingData([1, ...Array.from(report1Values)]);
```

Comment:
```suggestion
        TraktorS3.incomingData(new Uint8Array([1, ...Uint8Array.from(report1Values)]));
```
This should also be Uint8Array

Reply:
Fixed. The HID parser previously accepted both regular arrays and Uint8Array but after #4894 it only accepts Uint8Arrays.